signature/sign_test.go

```diff
@@ -134,6 +139,126 @@ func TestSignVerify(t *testing.T) {
 	}
 }
 
+var _ crypto.Signer = MockCryptoSigner{}
+
+type MockCryptoSigner struct {
+	privateKey crypto.PrivateKey
+	publickKey crypto.PublicKey
+}
+
+func (m MockCryptoSigner) Public() crypto.PublicKey { return m.publickKey }
+
+func (m MockCryptoSigner) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
+	return ecdsa.SignASN1(rand, m.privateKey.(*ecdsa.PrivateKey), digest)
+}
+
+func (m MockCryptoSigner) Algorithm() jwa.KeyAlgorithm {
+	return jwa.ES256
+}
+
+func TestSignVerifyCryptoSigner(t *testing.T) {
+
+	t.Parallel()
+	ctx := context.Background()
+
+	step := &pipeline.CommandStep{
+		Command: "llamas",
+		Plugins: pipeline.Plugins{
+			{
+				Source: "some-plugin#v1.0.0",
+				Config: nil,
+			},
+			{
+				Source: "another-plugin#v3.4.5",
+				Config: map[string]any{"llama": "Kuzco"},
+			},
+		},
+		Env: map[string]string{
+			"CONTEXT": "cats",
+			"DEPLOY":  "0",
+		},
+	}
+	// The pipeline-level env that the agent uploads:
+	signEnv := map[string]string{
+		"DEPLOY": "1",
+	}
+
+	// The backend combines the pipeline and step envs, providing a new env:
+	verifyEnv := map[string]string{
+		"CONTEXT": "cats",
+		"DEPLOY":  "0",
+		"MISC":    "llama drama",
+	}
+
+	stepWithInvariants := &CommandStepWithInvariants{
+		CommandStep:   *step,
+		RepositoryURL: "fake-repo",
+	}
+
+	cases := []struct {
```

Comment:
Since there's only one test case, it's probably overkill to set up the table-driven test infrastructure like this, but OTOH might be useful if we plan to add more cases?

Reply:
@DrJosh9000 yeah I figure at some point we will need more test cases so i left it a table.